src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/advice/ApplicationProblemControllerAdviceTest.java

```diff
@@ -210,8 +211,11 @@ void should_handle_constraint_violation_exception() {
 
     @Test
     void should_handle_http_message_not_readable_exception() {
+        // Mocking the HttpInputMessage
+        HttpInputMessage mockHttpInputMessage = mock(HttpInputMessage.class);
+
         HttpMessageNotReadableException httpMessageNotReadableException =
-            new HttpMessageNotReadableException("someMessage");
+            new HttpMessageNotReadableException("someMessage", mockHttpInputMessage);
```

Comment:
Just out of curiosity was this causing a problem without the mock?

Reply:
The use of HttpMessageNotReadableException(string message) is deprecated https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/converter/HttpMessageNotReadableException.html

This was being flagged by sonarcloud.